rocksdb: 5.11.3 -> 6.1.2

magenbluten opened the pull request:
###### Motivation for this change

rocksdb was out of date. also removed stuff that seemed broken.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
maintainers: add magenbluten

ryantm commented:
cc @adevress

ryantm commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,41 @@
-  tools = [ "sst_dump" "ldb" "rocksdb_dump" "rocksdb_undump" "blob_dump" ];
-in
 stdenv.mkDerivation rec {
   name = "rocksdb-${version}";
```
```suggestion
  pname = "rocksdb";
```

ryantm commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,41 @@
 
   src = fetchFromGitHub {
     owner = "facebook";
     repo = "rocksdb";
```
```suggestion
    repo = pname;
```

ryantm commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,41 @@
-    "install-static"
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ lz4 zlib zstd bzip2 ];
+  checkInputs = [ ];
```
Is `checkInputs` needed? Does cmake automatically do some kind of checking? If we need to turn of checking, can we use `doCheck = false;` instead?

ryantm commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,41 @@
-    install -D ${stdenv.lib.concatStringsSep " " tools} ''${!outputBin}/bin
-  '';
-
   enableParallelBuilding = true;
```
cmake automatically enables parallel building
```suggestion
```

ryantm commented:
@GrahamcOfBorg build rocksdb

ryantm commented on lib/licenses.nix:
```diff
@@ -50,6 +50,11 @@ lib.mapAttrs (n: v: v // { shortName = n; }) rec {
     free = false;
   };
 
+  apache20 = spdx {
```
This license is already in the file as `asl20`.

magenbluten replied:
sorry. missed that.

Mic92 commented on lib/licenses.nix:
```diff
@@ -50,6 +50,11 @@ lib.mapAttrs (n: v: v // { shortName = n; }) rec {
     free = false;
   };
 
+  apache20 = spdx {
```
Please also remove this.

Mic92 reviewed (approved):
Looks good to me otherwise.

Mic92 commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,42 @@
+    (lib.optional
+        (stdenv.hostPlatform.system == "i686-linux"
+         || stdenv.hostPlatform.system == "x86_64-linux")
+        "-DFORCE_SSE42=1")
```
Could this break somewhere?

Commit:
rocksdb: 5.11.3 -> 6.1.2

- mark osquery as broken
- mark ceph as broken

both osquery and ceph packages are outdated. furthermore, ceph has its
own inline rocksdb source tree which isn't use in the current nixpkg.
this needs to be fixed.

Ma27 commented on pkgs/tools/system/osquery/default.nix:
```diff
@@ -106,5 +106,6 @@ stdenv.mkDerivation rec {
     license = licenses.bsd3;
     platforms = platforms.linux;
     maintainers = with maintainers; [ cstrahan ma27 ];
+    broken = true;
```
What exactly broke `osquery`'s build here? And have you checked if there are some upstream patches to work around the problem?

magenbluten replied:
actually, the osquery nixpkg is quite outdated. a plugin of osquery ( in the current source ) depends on rocksdb-lite. the plugin expects the lite version of rocksdb to be linkable via -lrocksdb_lite. but since i didn't have time and i didn't understand the osquery nixpkg ( looks quite complicated ). hence, i decided to mark osquery broken. i need to be repackaged anyways.

Mic92 replied:
@flokli maybe regarding osquery?

Ma27 replied:
In fact I'm on of the maintainers of osquery, that's the original reason why I asked.
In the end @magenbluten is right, osquery is behind upstream by at least one minor. I tried to do the bump last month (hoping that this would solve several issues), but I faced some hard-to-solve erors back then. I hope that I have sufficient time soon to do the bump.

magenbluten replied:
@Ma27 if you could disable the rocksdb plugin or `fix` the CMakefile to link against `-lrocksdb` and not `-lrocksdb_lite` everything should work as expected. However, the rocksdb plugin says it wants rocksdb 5.7.2 afair, this is also outdated. the current release of osquery is 3.3.2 whereas the nixpkg version is 3.2.9.

Ma27 replied:
Hmm I see. Admittedly I'm slightly hesitant to patch the CMake files even further as it diverges pretty much from upstream already.

Due to `osquery` being slightly outdated and causing issues with further packages I guess that it should be fine for now to mark it as broken. I already have a local branch where I tried to bump/repackage osquery, but it's far from done. I hopefully have sufficient time for some more hacking next week, I can give this a try then :)

Lassulus commented on pkgs/top-level/all-packages.nix:
```diff
@@ -12621,7 +12621,7 @@ in
-  rocksdb = callPackage ../development/libraries/rocksdb { jemalloc = jemalloc450; };
+  rocksdb = callPackage ../development/libraries/rocksdb { };
 
   rocksdb_lite = rocksdb.override { enableLite = true; };
```
I guess you can remove this also, since this isn't supported anymore

Lassulus commented:
@GrahamcOfBorg build ceph

Lassulus commented:
this also breaks ceph. and everything that depends on it. But ceph seems to be an old version? @adevress  @alexanderkjeldaas

Lassulus commented:
since no one responded for the ceph breakage, we marked ceph as broken.

flokli commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,42 @@
+  buildInputs = [ bzip2 lz4 snappy zstd ];
+
+  cmakeFlags = [
+    "-DCMAKE_BUILD_TYPE=Release"
```
already set by `pkgs/development/tools/build-managers/cmake/setup-hook.sh`.

flokli commented on pkgs/development/libraries/rocksdb/default.nix:
```diff
@@ -1,86 +1,42 @@
+    "-DWITH_BZ2=1"
+    "-DWITH_LZ4=1"
+    "-DWITH_SNAPPY=1"
+    "-DWITH_ZLIB=0"
```
why was this not enabled?

magenbluten replied:
for some reason cmake could find zlib. didn't know how to fix it due to lack of time. but should be figured out why at some point. did work before v6.

flokli reviewed (commented):
some nitpicks.

magenbluten commented:
i tried to unbreak osquery with no luck. i failes when linking against rocksdb(-lite). see https://gist.github.com/magenbluten/997142eaa828ccd177c0dd06cc8261b4

Lassulus commented:
@Ma27 any points against merging this? since you recently updated osquery and this PR would break it.

Ma27 commented:
No. While working on `osquery` quite recently I realized that they currently have a hard time staying up-to-date with new software versions and I don't consider `osquery` important enough to actually block this PR any longer.

Therefore, it's IMHO ok to merge this now, I'll take care of `osquery` on one of the next weekends :)

Ma27 commented:
@magenbluten @Lassulus I managed to get `osquery` building locally (didn't test `ceph` as a bunch of python2 deps appear to be broken on master atm) and it works perfectly fine when using the old build system (no cmake). I suspect this is because the cmake build doesn't create statically linked libraries.

According to the [install section](https://github.com/facebook/rocksdb/blob/master/INSTALL.md) it's recommended to simply use `make {shared,static}_lib` depending on what's needed.

I'd simply go back to the old build approach for rocksdb (and keep 6.1.2), however I'm not sure if I'm missing a reason why cmake is used now (and I probably just didn't manage to get the static lib building with cmake ^^).

Do you think using the old make-based approach makes sense or am I missing something? :)

magenbluten commented:
i guess the install section you mention is outdated (dez. 2018). 

the "correct" solution would be to figure out how to make cmake make static libs ;)

magenbluten commented:
git grep -i static_lib:

CMakeLists.txt:set(ROCKSDB_STATIC_LIB rocksdb${ARTIFACT_SUFFIX})
